tests/test_cases/test_cocotb/test_cocotb.py

```diff
@@ -372,10 +372,9 @@ def do_test_afterdelay_in_readonly(dut, delay):
     exited = True
 
 
-@cocotb.test(expect_error=True,
+@cocotb.test(expect_error=(not cocotb.SIM_NAME.lower().startswith(("modelsim",))),
```

Comment:
```suggestion
@cocotb.test(expect_error=(cocotb.result.SimFailure if cocotb.SIM_NAME.lower().startswith(("modelsim",)) else False),
```

Reply:
Perhaps this needs more thought. Failing is doing the wrong thing, should we adjust all the simulators to fail in this case?